design/accepted/gc_plus/uncomitted-gc.md

```diff
@@ -79,19 +79,26 @@ objects added using this operation are never collected by GC.
 
 #### S3 Gateway CopyObject
 
-1. Copy object in the same branch will work the same - creating a new staging entry using the existing entry information.
-2. For objects that are not part of the branch, use the underlying adapter copy operation.
+When performing a shallow copy - track copied objects in ref-store.
+GC will read the copied objects information from the ref-store, and will add them to the list of uncommitted.
+lakeFS will periodically clear the copied list according to timestamp.
+
+1. Copy of staged object in the same branch will perform a shallow copy as described above
+2. All other copy operations will use the underlying adapter copy operation.
 
 #### Move/RenameObject
 Clients working through the S3 Gateway can use the CopyObject + DeleteObject to perform a Rename or Move operation.
 For clients using the OpenAPI this could have been done using StageObject + DeleteObject.
 To continue support of this operation, introduce a new API to rename an object which will be scoped to a single branch.
+Rename will add copied objects in ref-store similarly to CopyObject 
```

Comment:
So it needs to maintain that new table too, right?

Reply:
Added